api/src/api/invites.ts

```diff
@@ -0,0 +1,395 @@
+/*
+ * Copyright 2021, 2022 Macquarie University
+ *
+ * Licensed under the Apache License Version 2.0 (the, "License");
+ * you may not use, this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing software
+ * distributed under the License is distributed on an "AS IS" BASIS
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND either express or implied.
+ * See, the License, for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Filename: invites.ts
+ * Description:
+ *   This module contains invite related API routes at /api/invites
+ */
+
+import {
+  Action,
+  GetInviteByIdResponse,
+  GetProjectInvitesResponse,
+  GetTeamInvitesResponse,
+  isAuthorized,
+  PostCreateInviteInputSchema,
+  PostCreateProjectInviteResponse,
+  PostCreateTeamInviteResponse,
+  projectInviteToAction,
+  Resource,
+  teamInviteToAction,
+} from '@faims3/data-model';
+import express, {Response} from 'express';
+import {z} from 'zod';
+import {processRequest} from 'zod-express-middleware';
+import {
+  createInvite,
+  deleteInvite,
+  getInvite,
+  getInvitesForResource,
+  isInviteValid,
+} from '../couchdb/invites';
+import * as Exceptions from '../exceptions';
+import {requireAuthenticationAPI} from '../middleware';
+import patch from '../utils/patchExpressAsync';
+
+// This must occur before express api is used
+patch();
+
+export const api = express.Router();
```

Comment:
Hmm, I don't suppose the swagger.json documentation has been updated to align with the newer API routes.  We should update it or find another way to generate similar documentation.

Reply:
defer - there is an existing ticket for this